tensorflow/contrib/learn/python/learn/datasets/base.py

```diff
@@ -136,11 +136,12 @@ def maybe_download(filename, work_directory, source_url):
     gfile.MakeDirs(work_directory)
   filepath = os.path.join(work_directory, filename)
   if not gfile.Exists(filepath):
-    with tempfile.NamedTemporaryFile() as tmpfile:
+    with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
       temp_file_name = tmpfile.name
       urllib.request.urlretrieve(source_url, temp_file_name)
```

Comment:
Do we need to close `tmpfile` as well before doing the `urlretrieve`? (I assume this is file-locking related...)